### PR TITLE
[feat] Add okx

### PR DESCRIPTION
**Pull request by colmazia**
This is my first time, please be gentle

## Implementation details
- Do 1
- Do 2

## Please ensure the following requirements are met before submitting a pull request:

- [ ] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the GitHub PR explorer)

### CODE REVIEW

**Review by RogerKSI — CHANGES_REQUESTED** [review]

File: bothan-okx/src/service.rs [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+    //  resubscribe to all ids
+    let keys = cache.keys().await;
+
+    // If the keys are empty, subscribe to a single symbol to keep the connection alive
+    // TODO: find a better solution
+    let cmd = match keys.is_empty() {
+        true => Command::Subscribe(vec!["BTC-USDT".to_string()]),
+        false => Command::Subscribe(keys),
+    };
```
Comment (RogerKSI) [inline bothan-okx/src/service.rs:170]: does OKX also require a subscription to keep a connection like others?

Comment (colmazia) [reply bothan-okx/src/service.rs:170]: yes

File: bothan-api/server/src/main.rs [hunk 2]
```diff
@@ -12,6 +12,7 @@ use bothan_coinmarketcap::CoinMarketCapServiceBuilder;
 use bothan_cryptocompare::CryptoCompareServiceBuilder;
 use bothan_htx::HtxServiceBuilder;
 use bothan_kraken::KrakenServiceBuilder;
+use bothan_okx::OKXServiceBuilder;
```
[the remark this answers is not in the document]
Comment (colmazia) [reply bothan-api/server/src/main.rs:22]: or should we change Htx to HTX

**Review by warittornc — APPROVED** [review]
please resolve prior comments. Otherwise lgtm.

File: bothan-api/server/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (colmazia) [reply bothan-api/server/src/main.rs]: fixed

**Review by RogerKSI — APPROVED** [review]

**Pull request merged**